var/spack/repos/builtin/packages/py-pykokkos-base/package.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+#
+# ----------------------------------------------------------------------------
+
+from spack import *
+
+
+class PyPykokkosBase(CMakePackage, PythonPackage):
+    """Minimal set of bindings for Kokkos interoperability with Python
+    (initialize, finalize, View, DynRankView, Kokkos-tools)"""
+
+    homepage = "https://github.com/kokkos/pykokkos-base.git"
+    git = "https://github.com/kokkos/pykokkos-base.git"
+    maintainers = ["jrmadsen"]
+
+    version("main", branch="main", submodules=False)
+    version(
+        "0.0.5", commit="45f6e892c007ab124fabb3a545f4744537eafb55", submodules=False
+    )
+    version(
+        "0.0.4", commit="2efe1220d0128d3f2d371c9ed5234c4978d73a77", submodules=False
+    )
+    version(
+        "0.0.3", commit="4fe4421ac624ba2efe1eee265153e690622a18a5", submodules=False
+    )
```

Comment:
Versions are omitted from style checks so it's typical to write each version as a single line for compactness/readability.

Reply:
black did this, I suspect bc it doesn't fit within 78 characters.

Reply:
Ah, I'm not familiar with `black`. I've only done style validation with `spack style` and then manually fixed the issues.

Reply:
spack style isn't reporting any issues

Reply:
Black isn't fully supported as part of the standard spack development workflow, though the other maintainers expect it to be this year. When it is, I think it will correctly allow `version` definitions to be on a single line even if they exceed 78 characters.

I'm informing you that even though the formatting is PEP-8 compliant, it's not the canonical way of writing a spack package:
```console
$ cd $SPACK_ROOT/var/spack/repos/builtin/packages
$ grep 'version($' */package.py | wc -l
      38
$ grep "version('" */package.py | wc -l
   19043
```

There are plenty of ways to write a package definition that technically passes the style checks but are nonstandard, like putting the function definitions before the variants or putting the versions between the homepage and description strings.

Reply:
Those lines are 89 characters if all put on one line. Your flake8 file will fail if the line exceeds 88: https://github.com/spack/spack/blob/50dac14d109398a0e46d9168a0d490ecf1a739d2/.flake8#L40

Reply:
I could theoretically remove the `submodules=False` to get it on one line but I'd rather keep it there since this package _does have submodules_ but they shouldn't be cloned.

Reply:
The flake8 formatting done by spack turns out to be substantially more complicated than just the top-level `.flake8` file: see https://github.com/spack/spack/pull/20376 which points to the `share/spack/qa/flake8_formatter.py`. Trust me that the formatting check will pass if you combine the versions into one per line. 🙂

Reply:
😩 fine. Although I'm doing it in protest re: my previous comments about patching things after running the formatter. I guess this isn't a file that will edited much so I'll quit being an unnecessary pain and sed the double quotes too

Reply:
Sorry for the trouble. I think spack packages tend toward "write once, read many" (or 'write once, read many' ha ha) so hopefully this won't impose much of a maintenance burden. Thanks!